### DISCUSSION TITLE
Autocomplete Events

### DISCUSSION THREAD

**Discussion by RealityMasque** [Megabit/Blazorise#5488] [Q&A]
Is there documentation somewhere that list what Autocomplete events (e.g., SelectedTextChanged, SelectedValueChanged, etc) are raised as a user types in characters & selects an items from the match list?

**Comment by stsrki** [comment]
This might help https://blazorise.com/docs/specifications/autocomplete

**Reply by RealityMasque** [reply]
yeah, found that, but I'm not seeing there which events fire specifically for "user types in character" vs "user selected 1 option from match list".

**Reply by stsrki** [reply]
@David-Moreira might help you more with this.

**Reply by David-Moreira** [reply]
If I remember correctly there are nuances. 
The value will be fired on commit, which I believe is what you would expect from a dropdown like selection.
The text may be fired on change if `FreeTyping` is enabled. Except for a few nuances, like the `FreeTyping` feature, these two should mostly be fired at similar times, again the Value one acting more like how a dropdown selection which means, when the user commits to the value or exits the `Autocomplete`.

There may be times where the `Autocomplete` deems these should be called to inform of the text or value being reset.

There are separate events to keep track of the actual text being searched that should be fired as the user types, if you have that requirement, I think `SearchTextChanged`.

**Reply by RealityMasque** [reply]
hmm...  I have FreeTyping=true.  Maybe more details on the requirements I have might help.  

There is a list of objects the user can pick from using the Autocomplete, where Name property of the objects is what is being matched in the Autocomplete.  When the user selects one from the match list, other fields need to be disabled.  When the user is not selecting one from the match list (including after having selected one), those other fields need to be enabled.

I was thinking I could use SearchChanged for the 2nd case (since the user isn't selecting a match, just changing what they are searching on), & for the 1st case I could use SelectedValueChanged (since the user is selecting a match).  When I tried this combo & tested selecting an item from the matches, SelectedValueChanged is raised (disable the other fields) then SearchChanged is raised (enabling the other fields).

Do that make sense?

**Reply by RealityMasque** [reply]
@David-Moreira does that help?

**Reply by David-Moreira** [reply]
Sorry I haven't been able to take the time to analyse your question yet. I'll see if I can go over it today.

**Reply by David-Moreira** [reply]
So it seems that your issue is maybe with the order that the events are triggered, even though they are apparently working as intended, is this correct?

For requirement 1, it seems like it's ok to listen to the SelectedValueChanged and raise the fields according to the selection.
For requirement 2, 
try one of this options : 
- Use `SearchTextChanged` instead of `SearchChanged` (yes I know, too many similar events, we'll probably cut down on these in the future, anyway the `SearchTextChanged` should be more faithful to whenever the underlying textbox triggers while the `SearchChanged` to how we keep track of the currently searched value internally, so it can trigger more times)
- Use `Opened` to enable to fields **forcefully** and `Closed` to remove the force enable?

Do you think any of this options would work for you?

**Reply by RealityMasque** [reply]
Not sure, but requirements have changed.  :)
